### ISSUE TITLE
bug(mat-chip): Accessibility issue: Interactive controls must not be nested 

### ISSUE THREAD

**Issue by tek-moglykisvergad** [angular/components#24605]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

When running the axe DevTools extension, any `mat-chip` that contains a button with `matChipRemove` has the following accessibility issue _Interactive controls must not be nested_.

The following screenshot gives more details on the issue:

![image](https://user-images.githubusercontent.com/51793402/158655895-1eedc3e4-11a9-48a7-afc6-808bb63ad07f.png)

Suggested solution:
<img width="633" alt="image" src="https://user-images.githubusercontent.com/51793402/158656320-ec95b3c7-d833-4f32-bc49-1b41586a52a5.png">

### Reproduction

Steps to reproduce:
1. Download the axe DevTools extension: https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US
2. Navigate to the angular material chips' examples: https://material.angular.io/components/chips/examples
3. Run the axe DevTools on the entire page

### Expected Behavior

No accessibility issues for `mat-chip` under "Serious"

### Actual Behavior

"Interactive controls must not be nested" accessibility issue

### Environment

- Angular: 13
- CDK/Material: 13
- Browser(s): Google Chrome Version 98.0.4758.102 (Official Build) (64-bit)
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by jelbourn** [comment]
This issue should be addressed in the MDC-based version of the chips, which splits the chips components into three variants: `role="grid"`, `role="listbox"`, and no role. The remove button should only be used with the `role="grid"` pattern in the new implementation.

**Comment by egarkavy** [comment]
Any update on this. May be some lifehacks which can take off the button in button and put them together in the container?

**Comment by jelbourn** [comment]
The MDC-based chips will be promoted to stable in v15
